qwen_agent/keygen_strategies/split_query.py

```
@@ -0,0 +1,90 @@
+                 system_message: Optional[str] = DEFAULT_SYSTEM_MESSAGE,
+                 **kwargs):
+        super().__init__(function_list, llm, system_message, **kwargs)
+        # Currently, instruction is not utilized,
+        # so in order to avoid generating redundant tokens, set 'instruction' as stop words
+        self.extra_generate_cfg = merge_generate_cfgs(
+            base_generate_cfg=self.extra_generate_cfg,
+            new_generate_cfg={'stop': ['"], "instruction":']},
+        )
+
+    def _run(self, messages: List[Message], lang: str = 'en', **kwargs) -> Iterator[List[Message]]:
+        for last in super()._run(messages=messages, lang=lang, **kwargs):
```

Ruff (0.8.2)

Loop control variable `last` not used within loop body

(B007)